main.cpp

```diff
@@ -1,33 +1,19 @@
 /* mbed Microcontroller Library
- * Copyright (c) 2018 ARM Limited
+ * Copyright (c) 2018-2019 ARM Limited
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include "mbed.h"
-#include "ThisThread.h"
-#include "stats_report.h"
+#if !MBED_BARE_METAL
+extern void blinky_rtos();
+#else
+extern void blinky_bare_metal();
+#endif // !MBED_BARE_METAL
 
-DigitalOut led1(LED1);
-
-#define SLEEP_TIME                  500 // (msec)
-#define PRINT_AFTER_N_LOOPS         20
-
-// main() runs in its own thread in the OS
 int main()
 {
-    SystemReport sys_state( SLEEP_TIME * PRINT_AFTER_N_LOOPS /* Loop delay time in ms */);
-
-    int count = 0;
-    while (true) {
-        // Blink LED and wait 0.5 seconds
-        led1 = !led1;
-        ThisThread::sleep_for(SLEEP_TIME);
-
-        if ((0 == count) || (PRINT_AFTER_N_LOOPS == count)) {
-            // Following the main thread wait, report on the current system status
-            sys_state.report_state();
-            count = 0;
-        }
-        ++count;
-    }
+#ifndef MBED_BARE_METAL
```

Comment:
Could we try to simplify it and maybe showcase that some of the RTOS APIs can be used both with and without rtos? I'd like the code for blinky not to go over 1 screen it has to be simple and clear. Your example for RTOS is nice, but it my opinion it fits more exactly as RTOS example. Also adding short explanation in README what is the difference between normal and bare metal modes with link to docs would be useful.

Reply:
Re-worked it including your feedback d13f269